db/sqlglue.c

```diff
@@ -6440,8 +6440,12 @@ int sqlite3BtreeCloseCursor(BtCursor *pCur)
     /* update cursor use counts.  don't lock for now.
      * analyze shouldnt' affect cursor stats */
     if (pCur->db && !clnt->is_analyze) {
-        if (pCur->ixnum != -1)
+        if (pCur->ixnum != -1) {
             pCur->db->sqlixuse[pCur->ixnum] += (pCur->nfind + pCur->nmove);
+            pCur->db->index_used_count++;
+        } else {
+            pCur->db->read_count += (pCur->nfind + pCur->nmove);
```

Comment:
I think that this will be more than the actual number of records read? Probably okay though.

Reply:
Good catch! The off-by-1 issue seems to be caused by double-counting in `nfind` and `nmove`.